### PR TITLE
Gpu compatibility v2

### PR DESCRIPTION
**Pull request by mtjhrc**
@dorindabassey

Todo list:

- The `set_gpu_socket` trait method currently takes a UnixStream as an argument, it should take a `GpuBackend`. The `GpuBackend` should work, but I haven't tested it yet. (The `GpuBackend` should be constructible from the `UnixStream`).
- The structs needed as replies for the `GpuBackend::send_message` are not defined yet. (There are probably identical to the ones in the virtio spec - so they are defined in our vhost-user-gpu so just copy them over. (Probably put them in the `gpu_message.rs` file).
- Define methods such as `GpuBackend::get_display_info` that calls `GpuBackend::send_message` with `GpuBackendReq::GET_DISPLAY_INFO` and correct `V` generic parameter (previous bullet point) (`V` is the struct that represent the reply format)
   - (this needs to be done for each item of `GpuBackendReq` enum)  
- Attempt to call  `GpuBackend::get_display_info` in our `vhost-user-gpu`.
- Some cleanup:
    - I think there could be 3 files for the messages: `gpu_message.rs`, `generic_message.rs` (maybe also come up with a better name) and `vhost_message.rs`  Basically put the trait definitions in the `generic_message.rs` and normal vhost messages in the `vhost_message.rs`. This would also mean undoing the commit "Move VhostUserMsgHeader to a separate file"

### CODE REVIEW

File: vhost/src/vhost_user/message.rs [hunk 1]
```diff
@@ -243,152 +243,6 @@ bitflags! {
...
-            return false;
-        } else if self.get_version() != 0x1 {
-            return false;
-        } else if (self.flags & VhostUserHeaderFlag::RESERVED_BITS.bits()) != 0 {
-            return false;
-        }
-        true
-    }
-}
-
```
Comment (dorindabassey) [inline vhost/src/vhost_user/message.rs:391]: Question: why did you move this to a new rust file? header.rs?

Comment (mtjhrc) [reply vhost/src/vhost_user/message.rs:391]: I was trying to separate the normal vhost and gpu vhost headers to separate files. But moving this didn't end up making sense in the end. It would be better to undo this change. (see also the last bullet point in the PR description)
